addons/minio/template/base/install.sh

```diff
@@ -247,7 +271,344 @@ function allow_pvc_resize() {
+    if [ -n "$MINIO_HOSTPATH" ]; then
+        kubectl delete -k "$dst/hostpath"
+        return
+    fi
+    kubectl delete -k "$dst/pvc"
+}
+
+# minio_swap_fs_migration_pvs swaps the pv backing the minio deployment with the pv used during the migration.
+function minio_swap_fs_migration_pvs() {
+    local minio_pv=$1
+    local migration_pv=$2
+    local src="$DIR/addons/minio/__MINIO_DIR_NAME__"
+    local dst="$DIR/kustomize/minio"
+
+    MINIO_ORIGINAL_CLAIM_SIZE=$(kubectl get pvc -n "$MINIO_NAMESPACE" minio-pv-claim --no-headers -o custom-columns=:.spec.resources.requests.storage)
+    if [ -z "$MINIO_ORIGINAL_CLAIM_SIZE" ]; then
+        MINIO_ORIGINAL_CLAIM_SIZE="$MINIO_CLAIM_SIZE"
+    fi
+
+    kubectl patch pv "$migration_pv" --type=json -p='[{"op": "remove", "path": "/spec/claimRef"}]'
+    kubectl delete pvc -n "$MINIO_NAMESPACE" minio-pv-claim
+    kubectl patch pv "$minio_pv" --type=json -p='[{"op": "remove", "path": "/spec/claimRef"}]'
+
+    printf "${YELLOW}A backup of the original Minio data has been stored in PersistentVolumeClaim minio-pv-claim-backup${NC}\n"
+
+    MINIO_NEW_VOLUME_NAME="$migration_pv"
+    MINIO_ORIGINAL_VOLUME_NAME="$minio_pv"
+    render_yaml_file_2 "$src/migrate-fs/minio-pvc.yaml" > "$dst/migrate-fs/minio-pvc.yaml"
+    kubectl create -f "$dst/migrate-fs/minio-pvc.yaml"
+    render_yaml_file_2 "$src/migrate-fs/minio-backup-pvc.yaml" > "$dst/migrate-fs/minio-backup-pvc.yaml"
+    kubectl create -f "$dst/migrate-fs/minio-backup-pvc.yaml"
+}
+
+# minio_swap_fs_migration_hostpaths moves the host path used during the migration to the host path used by the
+# original minio pod. a backup is kept under hostpath.bkp directory.
+function minio_swap_fs_migration_hostpaths() {
+    local suffix
+    local bkp_location
+    suffix=$(< /dev/urandom tr -dc A-Za-z0-9 | head -c8)
+    bkp_location=$(printf "%s-%s" "$MINIO_HOSTPATH" "$suffix")
+    mv "$MINIO_HOSTPATH" "$bkp_location"
+    printf "${YELLOW}A backup of the original Minio data has been stored at %s${NC}\n" "$bkp_location"
+
+    mv "$MINIO_MIGRATION_HOSTPATH" "$MINIO_HOSTPATH"
+    cp -Rfp "$bkp_location/.kurl" "$MINIO_HOSTPATH/"
+}
+
+# minio_uses_fs_format verifies if minio uses the legacy fs format. greps the file /data/.minio.sys/format.json
+# from within the pod.
+function minio_uses_fs_format() {
+    local format_string
+    format_string=$(kubectl exec -n $MINIO_NAMESPACE deploy/minio -- cat /data/.minio.sys/format.json 2>/dev/null)
+    if [ -z "$format_string" ]; then
+        bail "Failed to read /data/.miniosys/format.json inside minio pod"
+    fi
+    echo "$format_string" | grep -q '"format":"fs"'
+}
+
+# minio_svc_has_no_endpoints validates that the provided service in the provided namespace has no endpoints.
+function minio_svc_has_no_endpoints() {
+    local namespace=$1
+    local service=$2
+    kubectl get endpoints --no-headers -n "$namespace" "$service" -o custom-columns=:subsets | grep -q none
```

Comment:
we have a pattern of using -o jsonpath or --template. can you use that instead?

Reply:
fixed.